File: python/205.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        # check for distinct pairs
        arr = []
        for i in range(len(s)):
            if (s[i], t[i]) not in arr:
                arr.append((s[i], t[i]))
        
        if len(set(s)) == len(set(t)) == len(arr):
            return True
        else:
            return False

File: c++/205.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        std::set<char> set_s;
        std::set<char> set_t;
        std::set<tuple<int, int>> pairs;
        for(int i = 0; i < s.length(); i++) {
            set_s.insert(s[i]);
            set_t.insert(t[i]);
            tuple<int, int> tuple;
            tuple = make_tuple(s[i], t[i]);
            pairs.insert(tuple);
        }
    // std::cout << set_s.size() << set_t.size() << pairs.size();
    if (set_s.size() == set_t.size() && set_s.size() == pairs.size()) {
        return true;
    } else {
        return false;
    }
    }
};
